pom.xml

```diff
@@ -698,16 +698,6 @@
             <artifactId>logback-classic</artifactId>
             <scope>test</scope>
         </dependency>
-        <dependency>
-            <groupId>junit</groupId>
-            <artifactId>junit</artifactId>
-            <scope>test</scope>
-        </dependency>
-        <dependency>
-            <groupId>org.mockito</groupId>
-            <artifactId>mockito-core</artifactId>
-            <scope>test</scope>
-        </dependency>
```

Comment:
Those can be moved to a profile which can be activated by the presence or absence of a property in the maven project.  So only the `sshd-benchmarks` POM would have to define this property, but all other subprojects would contain the dependencies.

Reply:
I don't understand. The profile would have to be in the parent POM, but the property in the child POM. AFAIK that won't work.

Reply:
It will.  Profiles are inherited but evaluated in the context of each project.

Reply:
Doesn't work for me. Parent POM:
```
        <profile>
            <id>test-dependencies</id>
            <activation>
                <property>
                    <name>!projectWithoutTests</name>
                </property>
            </activation>
            <dependencies>
                <dependency>
                    <groupId>org.slf4j</groupId>
                    <artifactId>jul-to-slf4j</artifactId>
                    <scope>test</scope>
                </dependency>
                <dependency>
                    <groupId>ch.qos.logback</groupId>
                    <artifactId>logback-core</artifactId>
                    <scope>test</scope>
                </dependency>
                <dependency>
                    <groupId>ch.qos.logback</groupId>
                    <artifactId>logback-classic</artifactId>
                    <scope>test</scope>
                </dependency>
                <dependency>
                    <groupId>junit</groupId>
                    <artifactId>junit</artifactId>
                    <scope>test</scope>
                </dependency>
                <dependency>
                    <groupId>org.mockito</groupId>
                    <artifactId>mockito-core</artifactId>
                    <scope>test</scope>
                </dependency>
            </dependencies>
        </profile>
```
sshd-benchmarks POM then just defines the property:
```
    <properties>
        <!-- ... -->
        <projectWithoutTests>true</projectWithoutTests>
    </properties>
```
Result: build fails with a compilation error.

Reply:
You're right, my bad. I was really expecting that to work.

Reply:
Found another way that seems to work: leave all other poms unchanged, keep the junit dependency in the parent POM. In sshd-benchmarks, override it by adding an explicit dependency:
```
        <dependency>
            <groupId>junit</groupId>
            <artifactId>junit</artifactId>
            <scope>provided</scope>
        </dependency>
```
I actually had tried that first and it hadn't worked. I'll have re-verify on my main development machine.